packages/winpanda/extra/src/winpanda/core/rc_ctx.py

```diff
@@ -147,17 +148,27 @@ def _get_cluster_conf_items(self, escape):
                               fallback=cm_const.ZK_CLIENTPORT_DFT))
             for s in cluster_conf.sections() if s.startswith('master-node')
         ]
-        master_priv_ipaddr = mnode_cfg_items[0][0] if mnode_cfg_items else (
-            '127.0.0.1'
-        )
-        zk_client_port = mnode_cfg_items[0][1] if mnode_cfg_items else (
-            cm_const.ZK_CLIENTPORT_DFT
-        )
+        if mnode_cfg_items:
+            master_priv_ipaddr = mnode_cfg_items[0][0]
+            zk_client_port = mnode_cfg_items[0][1]
+            dsc_type = self._cluster_conf.get('discovery', {}).get('type')
+            if dsc_type == 'strict':
+                master_location = f'{master_priv_ipaddr}:{zk_client_port}'
+            else:
+                master_location = ",".join(
+                    [escape(f'{v[0]}:{v[1]}') for v in mnode_cfg_items])
```

Comment:
One possible value here is `static`, not `strict`.

When it is `static` we need to join the masters by commas.

Otherwise, if `dsc_type` is `master_http_loadbalancer` we use the value in `exhibitor_address` from the dcos_config

Reply:
[Configuration Reference](https://docs.d2iq.com/mesosphere/dcos/1.9/installing/oss/custom/configuration/configuration-parameters/) here i can see: "...If master_discovery is set to **master_http_loadbalancer**, then exhibitor_storage_backend cannot be set to **static**..." So my assuption was that only two options possible. Thanks for clarification will change this.